src/app/client/src/app/modules/workspace/components/data-driven/data-driven.component.ts

```diff
@@ -0,0 +1,258 @@
+   * It is kept true by default as at first when we comes
+   * to a page the loader should be displayed before showing
+   * any data
+	 */
+  showLoader = true;
+  /**
+* To show toaster(error, success etc) after any API calls
+*/
+  private toasterService: ToasterService;
+
+  /**
+* urlString for get url details
+*/
+  private urlString;
+  /**
+* contentType is creation type, fected from url
+*/
+  public contentType;
+  /**
+ * userForm name creation
+ */
+  public creationForm: FormGroup;
+  /**
+ * userProfile is of type userprofile interface
+ */
+  public userProfile: IUserProfile;
+  /**
+* Contains config service reference
+*/
+  public config: ConfigService;
+  /**
+ * To make inbox API calls
+ */
+  private editorService: EditorService;
+  /**
+  * To call resource service which helps to use language constant
+  */
+  public resourceService: ResourceService;
+  /**
+ * To call resource service which helps to use language constant
+ */
+  public userService: UserService;
+  /**
+ * To send activatedRoute.snapshot to routerNavigationService
+ */
+  public activatedRoute: ActivatedRoute;
+  /**
+  * loader message
+  */
+  loaderMessage: ILoaderMessage;
+
+  public frameworkService: FrameworkService;
+
+  public formService: FormService;
+
+  public formType = 'content';
+
+  public formAction = 'create';
+
+  public getFormFields: any;
+
+  public formFieldProperties: any;
+
+  public categoryMasterList: any;
+
+  public creationFormLable: string;
+
+  public exists: boolean;
+
+  public framework: String;
+
+
+
+  constructor(
+    activatedRoute: ActivatedRoute,
+    frameworkService: FrameworkService,
+    private router: Router,
+    resourceService: ResourceService,
+    toasterService: ToasterService,
+    editorService: EditorService,
+    userService: UserService,
+    config: ConfigService,
+    formService: FormService,
+    private _cacheService: CacheService
+  ) {
+    this.activatedRoute = activatedRoute;
+    this.resourceService = resourceService;
+    this.toasterService = toasterService;
+    this.editorService = editorService;
+    this.userService = userService;
+    this.config = config;
+    this.frameworkService = frameworkService;
+    this.formService = formService;
+    this.activatedRoute.url.subscribe(url => {
+      this.contentType = url[0].path;
+    });
+    this.creationFormLable = this.config.appConfig.contentCreateTypeLable[this.contentType];
+    // console.log('creationFormLable', this.creationFormLable);
+  }
+
+  ngOnInit() {
+
+    /**
+     * getMetaData is called to config the form data and framework data
+     */
+    this.getMetaData();
+
+    /***
+ * Call User service to get user data
+ */
+    this.userService.userData$.subscribe(
+      (user: IUserData) => {
+        if (user && !user.err) {
+          this.userProfile = user.userProfile;
+        }
+      });
+  }
+  /**
+* getMetaData is gives form config data
+*/
+  getMetaData() {
```

Comment:
refer filter review comment for this component. After changes will continue with review